refactor(dRICH): compatibility update for PDU readout

c-dilks opened the pull request:
### Briefly, what does this PR introduce?

Updates respecting https://github.com/eic/epic/pull/331 

- `richgeo`:
  - use DD4hep `VariantParameters` to streamline access to sensor positioning and orientation
  - various updates respecting https://github.com/eic/epic/pull/331 
    - tests of sensor positioning and orientation are moved to `epic` with that PR
  - respect the readout field change: `module` -> `pdu` and `sipm`
  - method `richgeo::ReadoutGeo::PixelGapMask` was cut and pasted from `PhotoMultiplierHitDigi` 
- `PhotoMultiplierHitDigi`:
  - move pixel gap (mask) cuts to `richgeo` service in `ReadoutGeo`
  - delete config parameter `PhotoMultiplierHitDigiConfig::pixelSize`, since it can be obtained from `richgeo`


### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [x] Other: refactor

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
yes: this PR should be merged at the same time as
- https://github.com/eic/epic/pull/331
- https://github.com/eic/drich-dev/pull/83

### Does this PR change default behavior?
not really, benchmarks appear to be the same

c-dilks commented on src/algorithms/digi/PhotoMultiplierHitDigi.cc:
```diff
@@ -277,67 +266,12 @@ bool  eicrecon::PhotoMultiplierHitDigi::qe_pass(double ev, double rand) const
 // add a hit to local `hit_groups` data structure
 void eicrecon::PhotoMultiplierHitDigi::InsertHit(
     std::unordered_map<CellIDType, std::vector<HitData>> &hit_groups,
     CellIDType       id,
```
`InsertHit` is private, which hopefully reduces the chance of such mistakes.

wdconinc replied:
Can you nolint it? https://clang.llvm.org/extra/clang-tidy/#suppressing-undesired-diagnostics

c-dilks replied:
Like this?
```diff
 // add a hit to local `hit_groups` data structure
+// NOLINTBEGIN(bugprone-easily-swappable-parameters)
 void eicrecon::PhotoMultiplierHitDigi::InsertHit(
     std::unordered_map<CellIDType, std::vector<HitData>> &hit_groups,
     CellIDType       id,
     double           amp,
     TimeType         time,
     std::size_t      sim_hit_index,
     bool             is_noise_hit
-    )
+    ) // NOLINTEND(bugprone-easily-swappable-parameters)
 {
```
Or should I do this to the header file?

wdconinc replied:
I think that should do it.

wdconinc reviewed (approved):
Finally had a chance to review this again. Looks good.

wdconinc commented:
@c-dilks @chchatte92 This should be ready for merging, along with https://github.com/eic/epic/pull/331. Are you still ok with doing that, or do you need to complete https://github.com/eic/drich-dev/pull/83 first?

c-dilks commented:
Yes, they can be merged. https://github.com/eic/drich-dev/pull/83 is also ready, I will merge that one later tomorrow after re-reviewing (`drich-dev` `main` will break for a day, but I doubt anyone besides @chchatte92 will notice).

This PDU update worked fine a couple weeks ago, but now something appears to be wrong with the track propagation, so I can no longer check the dRICH benchmarks. All I know so far is that we no longer see `TrackPoints` from `RichTrack_factory`. I'll open an issue after some more testing.

wdconinc commented:
Not so surprising that RichTrack_factory stopped working. I assume that https://github.com/eic/EICrecon/blob/00fa2637e30cb5064a1034f98e0d28c952052504/src/detectors/DRICH/DRICH.cc#L113 relies on those CentralCKFTrajectories being of the Acts variety. Try using this variety: https://github.com/eic/EICrecon/blob/main/src/global/tracking/tracking.cc#L67.

wdconinc commented:
https://github.com/eic/EICrecon/pull/860 for details.

chchatte92 commented:
Hello colleagues, a week back, we have obtained the Nsigma separation for PDUs and everything seemed normal. 
[Aerogel+Gas_eta2.pdf](https://github.com/eic/EICrecon/files/12473486/Aerogel%2BGas_eta2.pdf)
[Aerogel+Gas_eta0.pdf](https://github.com/eic/EICrecon/files/12473488/Aerogel%2BGas_eta0.pdf)
[Aerogel+Gas_eta1.pdf](https://github.com/eic/EICrecon/files/12473490/Aerogel%2BGas_eta1.pdf)

chchatte92 commented:
@wdconinc will it be sufficient if I switch to `main` and check if everything works or not?

c-dilks commented:
> @wdconinc I switched to EICRecon main, changed the {"CentralCKFTrajectories"}, to {"CentralCKFActsTrajectories"} inside DRICH.cc but I get the following errors at compilation:
> 
> ```
> gpfs/compass/chchatte/EPIC/stable/drich-dev/EICrecon/src/algorithms/tracking/CKFTracking.cc:251:32: error: ‘class edm4eic::MutableTrajectory’ has no member named ‘addToTrackParameters’; did you mean ‘getTrackParameters’?
>   251 |                     trajectory.addToTrackParameters(pars);
>       |                                ^~~~~~~~~~~~~~~~~~~~
>       |                                getTrackParameters
> ```

I could not find your PR for this, hence testing with https://github.com/eic/EICrecon/pull/880.

chchatte92 commented:
@c-dilks hi Chris, I am sorry; I forgot to create a PR for this.  But I checked your PR and with that change one can have the track resolutions etc.
